src/terminal/terminal/terminal.h

```diff
@@ -915,261 +536,86 @@ void guac_terminal_scroll_handler(guac_terminal_scrollbar* scrollbar, int value)
+ * size are taken into account, and rows which are within the buffer but
+ * unavailable due to being outside the desired scrollback range are ignored.
  *
  * @param term
- *     The terminal which should receive the given data on STDIN.
- *
- * @param data
- *     The data the terminal should receive on STDIN.
- *
- * @param length
- *     The size of the given data, in bytes.
+ *     The terminal whose off-screen row count should be determined.
  *
  * @return
- *     The number of bytes written to STDIN, or a negative value if an error
- *     occurs preventing the data from being written. This should always be
- *     the size of the data given unless data is intentionally dropped.
+ *     The number of rows within the buffer which are not currently displayed
+ *     on screen.
  */
-int guac_terminal_send_data(guac_terminal* term, const char* data, int length);
+int guac_terminal_get_available_scroll(guac_terminal* term);
 
 /**
- * Sends the given string as if typed by the user. If terminal input is
- * currently coming from a stream due to a prior call to
- * guac_terminal_send_stream(), any input which would normally result from
- * invoking this function is dropped. 
+ * Returns the height of the given terminal, in characters.
  *
  * @param term
- *     The terminal which should receive the given data on STDIN.
- *
- * @param data
- *     The data the terminal should receive on STDIN.
+ *     The terminal whose height should be determined.
  *
  * @return
- *     The number of bytes written to STDIN, or a negative value if an error
- *     occurs preventing the data from being written. This should always be
- *     the size of the data given unless data is intentionally dropped.
+ *     The height of the terminal, in characters.
  */
-int guac_terminal_send_string(guac_terminal* term, const char* data);
+int guac_terminal_get_rows(guac_terminal* term);
 
 /**
- * Sends data through STDIN as if typed by the user, using the format string
- * given and any args (similar to printf). If terminal input is currently
- * coming from a stream due to a prior call to guac_terminal_send_stream(), any
- * input which would normally result from invoking this function is dropped.
+ * Returns the width of the given terminal, in characters.
  *
  * @param term
- *     The terminal which should receive the given data on STDIN.
- *
- * @param format
- *     A printf-style format string describing the data to be received on
- *     STDIN.
- *
- * @param ...
- *     Any srguments to use when filling the format string.
+ *     The terminal whose width should be determined.
  *
  * @return
- *     The number of bytes written to STDIN, or a negative value if an error
- *     occurs preventing the data from being written. This should always be
- *     the size of the data given unless data is intentionally dropped.
- */
-int guac_terminal_sendf(guac_terminal* term, const char* format, ...);
```

Comment:
IMHO, this should stay in public as a convenient printf-style way to invoke `guac_terminal_send_string()`, assuming there isn't anything about this function that would mean it has to be considered internal.

Reply:
Sure

Reply:
Done.